community/shell/src/main/java/org/neo4j/shell/StartClient.java

```diff
@@ -504,4 +500,29 @@ private static String pad( String string, int length )
         }
         return string;
     }
+
+    private static class ShellExecutionFailureException extends RuntimeException
+    {
+        private final Throwable cause;
+        private final Args args;
+
+        ShellExecutionFailureException( Throwable cause, Args args )
+        {
+            this.cause = cause;
+            this.args = args;
+        }
+
+        void dumpMessage( PrintStream out, PrintStream err )
+        {
+            String message = cause.getCause() instanceof ConnectException ?
+                             "Connection refused" : cause.getMessage();
+            err.println( "ERROR (-v for expanded information):\n\t" + message );
+            if ( args.has( "v" ) )
```

Comment:
Both `cause` and `args` are owned by the exception. Sure, they are given by `StartClient`, and `StartClient` is the only class using this exception, but I don't really see the problem as all resources needed for the message logic is contained within the exception class. Do you want to send `args` as a parameter to the method instead?